### ISSUE TITLE
Qwen8B at checkpoint-100 only Average@12: 73.06%

### ISSUE THREAD

**Issue by Guozhenyuan** [siyan-zhao/OPSD#17]
I use `/home/llm_data_3/OPSD/scripts/run_opsd_4b.sh` to train the model and evaluate with `/home/llm_data_3/OPSD/eval/run_eval.sh`.

I cannot reproduce the accuracy reported in the paper. (76.4%)

accelerate launch \
    --config_file accelerate.yaml \
    --num_processes 8 \
    --gradient_accumulation_steps 1 \
    --main_process_port 12949 \
    opsd_train.py \
    --model_name_or_path Qwen/Qwen3-4B \
    --learning_rate 5e-6 \
    --max_grad_norm 0.1 \
    --per_device_train_batch_size 4 \
    --gradient_checkpointing \
    --gradient_accumulation_steps 1 \
    --output_dir  ./output/opsd/ \
    --run_config qwen34b_gen1024_fixteacher_temp11_forwardbeta0_clip005 \
    --num_train_epochs 30 \
    --max_completion_length 1024 \
    --save_steps 25 \
    --logging_steps 2 \
    --attn_implementation flash_attention_2 \
    --torch_dtype bfloat16 \
    --max_length 20000 \
    --beta 0 \
    --use_vllm \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.6 \
    --vllm_tensor_parallel_size 1 \
    --use_peft \
    --lora_r 64 \
    --lora_alpha 128 \
    --lora_target_modules q_proj k_proj v_proj o_proj gate_proj up_proj down_proj \
    --temperature 1.1 \
    --top_p 0.95 \
    --top_k 20 \
    --lmbda 1 \
    --fixed_teacher \
    --jsd_token_clip 0.05 \
    --wandb_project OPSD

**Comment by siyan-zhao** [comment]
Hi, you can refer to our results here for references, hope this helps: https://github.com/siyan-zhao/OPSD/issues/11#issuecomment-4330688127

**Comment by Guozhenyuan** [comment]
Could you provide more results on the larger baselines and compare with other methods, such as GPRO et al.?

**Comment by siyan-zhao** [comment]
Sorry for the late reply. I do not have access to the detailed per step baseline scores right now due to switching clusters. I remember for GRPO and SFT, later steps tend to collapse when training on this reasoning dataset, especially for 1.7b model. SFT declines a lot due to the solution being too concise for SFT and since we are all evaluating in thinking mode. 
Hope this help!
